pkg/apis/core/types_binding.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:skipVerbs=get,list,create,update,patch,delete,deleteCollection,watch
```

Comment:
Remove this? Is it needed?

Reply:
Yes, otherwise generating List() function will complaint that `BindingList` is not defined and so on.

Reply:
Why do we need `List()` function? 👀 I don't see this `genclient` for Kubernetes' `Binding` type - why do we need it?

Reply:
OK. How is k/k preventing this? I don't see them using `// +genclient` for the `Binding` struct?

Reply:
I think it's not on by default, 
https://github.com/kubernetes/kubernetes/blob/9ad41812e58c12dc9666b1172d276c46a70aa5c5/hack/update-codegen.sh#L61-L66

It generates the clients based on types in staging/src.
and it generates the clients only for structs having the // +genclient
https://github.com/kubernetes/kubernetes/blob/9ad41812e58c12dc9666b1172d276c46a70aa5c5/staging/src/k8s.io/api/core/v1/types.go#L3933-L3939
will lead to https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod.go
The clients used in k/k are imported, because the output package mentioned here is k8s.io/client-go
https://github.com/kubernetes/kubernetes/blob/9ad41812e58c12dc9666b1172d276c46a70aa5c5/hack/update-codegen.sh#L80

Clients are not generated for types which dont have this tag,
eg: 
For ephemeralContainer, https://github.com/kubernetes/kubernetes/blob/9ad41812e58c12dc9666b1172d276c46a70aa5c5/staging/src/k8s.io/api/core/v1/types.go#L3809
there are no clients in https://github.com/kubernetes/client-go/tree/master/kubernetes/typed/core/v1

Atleast, this is what I could infer. Please correct me if I'm wrong.
@timebertt, Maybe you have some pointers?